opentimelineview/timeline_widget.py

```diff
@@ -537,6 +544,186 @@ def wheelEvent(self, event):
+                for selectedItem in self.scene().selectedItems():
+                    selectedItem.setSelected(False)
+            newSelectedItem.setSelected(True)
+            self.centerOn(newSelectedItem)
+
+    def _get_new_item(self, key_event, curSelectedItem):
+        key = key_event.key()
+
+        if key in (
+               QtCore.Qt.Key_Left,
+               QtCore.Qt.Key_Right,
+               QtCore.Qt.Key_Up,
+               QtCore.Qt.Key_Down,
+               QtCore.Qt.Key_Return,
+               QtCore.Qt.Key_Enter
+               ):
+            if key == QtCore.Qt.Key_Left:
+                newSelectedItem = self._get_left_item(curSelectedItem)
+            elif key == QtCore.Qt.Key_Right:
+                newSelectedItem = self._get_right_item(curSelectedItem)
+            elif key == QtCore.Qt.Key_Up:
+                newSelectedItem = self._get_up_item(curSelectedItem)
+            elif key == QtCore.Qt.Key_Down:
+                newSelectedItem = self._get_down_item(curSelectedItem)
+            elif key in [QtCore.Qt.Key_Return, QtCore.Qt.Key_Return]:
+                if isinstance(curSelectedItem, NestedItem):
+                    curSelectedItem.keyPressEvent(key_event)
+                    newSelectedItem = curSelectedItem
+        else:
+            newSelectedItem = curSelectedItem
+
+        return newSelectedItem
+
+    def keyPressEvent(self, key_event):
+        super(CompositionView, self).keyPressEvent(key_event)
+        self.setInteractive(True)
+
+        # No item selected, so select the first item
+        if len(self.scene().selectedItems()) <= 0:
+            newSelectedItem = self._get_first_item()
+        # Based on direction key, select new selected item
+        else:
+            curSelectedItem = self.scene().selectedItems()[0]
+
+            # Check to see if the current selected item is a rect item
+            # If current selected item is not a rect, then extra tests
+            # are needed.
+            if not isinstance(curSelectedItem, QtWidgets.QGraphicsRectItem):
+                if curSelectedItem.parentItem():
+                    curSelectedItem = curSelectedItem.parentItem()
+                    newSelectedItem = self._get_new_item(
+                                                         key_event,
+                                                         curSelectedItem
+                                                         )
+                else:
+                    newSelectedItem = self._get_new_item(
+                                                         key_event,
+                                                         curSelectedItem
+                                                         )
+            else:
+                newSelectedItem = self._get_new_item(
+                                                     key_event,
+                                                     curSelectedItem
+                                                     )
```

Comment:
The `newSelectedItem = ...` part seems the same in all three code paths, so if you pull it out of the if/else block then you'll only need it once.

Reply:
Yup, can take care of that as well.